.github/actions/artifact_download/action.yml

```diff
@@ -0,0 +1,30 @@
+name: Download artifact
+description: Download and decrypt an artifact.
+inputs:
+  name:
+    description: 'The name of the artifact.'
+    required: true
+  path:
+    description: 'Download to a specified path.'
+    required: false
+    default: ./
+  encryption-secret:
+    description: 'The secret to use for decrypting the artifact.'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install unzip
+      uses: montudor/action-zip@v1
+    - name: Create temporary directory
+      id: tempdir
+      shell: bash
+      run: echo "directory=$(mktemp -d)" >> "$GITHUB_OUTPUT"
+    - name: Download the artifact
+      uses: actions/download-artifact@v3
+      with:
+        name: ${{ inputs.name }}
+        path: ${{ steps.tempdir.outputs.directory }}
+    - name: Decrypt and unzip archive
+      shell: bash
+      run: unzip -P '${{ inputs.encryption-secret }}' -o -qq -d ${{ inputs.path }} ${{ steps.tempdir.outputs.directory }}/archive.zip
```

Comment:
I don't know if we actually want to overwrite (`-o`) here. Why did you choose to do so?

Reply:
Oh, this was an artifact of some tests I ran at the beginning and I overlooked it when I cleaned up the branch. I removed it now.